### PR TITLE
Update free-programming-books-langs.md

### PR DESCRIPTION
**Pull request by Abinashbunty**
Added Q# lanaguge used for developing quantum algorithms on Azure

## What does this PR do?
Add resource(s) | Improve repo

It adds new resource as well as improves repo in a sense that it added a completely new language which is in demand. 

### Description
The added resource is related to Q# language which is used to develop quantum algorithms on Azure.

### Why is this valuable (or not)?
The resource is valuable because quantum computing is gradually advancing and there is a need to implement new and advanced quantum algorithms. Q# language helps facilitate the same and we need more open-source users to learn and develop quantum algorithms.

### How do we know it's really free?
The source is directly from the Microsoft Learn platform which provides documentation of all Microsoft services. Here's the link for reference as well. [Q#](https://learn.microsoft.com/en-us/azure/quantum/user-guide/)

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: books/free-programming-books-langs.md [hunk 1]
```diff
@@ -172,6 +172,7 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
     * [Pandas](#pandas)
     * [Pyramid](#pyramid)
     * [Tornado](#tornado)
+* [Q#](#q)
```
[the remark this answers is not in the document]
Comment (Abinashbunty) [reply books/free-programming-books-langs.md]: Made the necessary changes.

**Comment by Abinashbunty** [comment]
@eshellman I think the change has been fixed. 😄

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Comment by Abinashbunty** [comment]
> Thanks, and welcome to free-programming-books!

Thank you. 😃

**Pull request merged**